### PR TITLE
Add eng/scripts/DirStructureStats.ps1

### PR DESCRIPTION
**Pull request by konrad-jamrozik**

### CODE REVIEW

File: eng/scripts/DirStructureStats.ps1
Comment (mikeharder) [inline eng/scripts/DirStructureStats.ps1:1]: Should we do this in pwsh or node?  I'd lean towards node.

**Review by mikeharder — DISMISSED** [review]
Should we do this in pwsh or node? I'd lean towards node.

**Comment by weshaggard** [comment]
What is the context for these?

**Comment by konrad-jamrozik** [comment]
@mikeharder @weshaggard my intent with these scripts was to share a "starting point" scripts people can use for manual analysis. Like: you dot-source the functions I shared, run them, get the table of results, and do manual post-processing by hand with search-and-replace in notepad++ or whatever (that's how I do it). This is also the reason for using PowerShell, not Node - this is supposed to be easily runnable by hand. Of course I can keep these scripts to myself only, but I figured I might as well share them with the team. 

Given this context, do have a better idea how to share them?

**Comment by weshaggard** [comment]
I don't have any problems including them here but I'm mostly just wanting us to try and get to a point where we can easily have shared code that understands the discovery mechanism of specs.

**Comment by konrad-jamrozik** [comment]
I'll do the scripts consolidation as part of:
- https://github.com/Azure/azure-sdk-tools/issues/8203

**Comment by mikeharder** [comment]
> Given this context, do have a better idea how to share them?

I think it's premature to add them to the repo, until we have a better idea who else will be using them, and how often.  For now, I would share them via another artifact in GitHub, like a closed issue in this repo, or a Public Gist.

**Comment by konrad-jamrozik** [comment]
I'll convert this to Gist. Thanks!

**Pull request closed without merging**